0020-py-version-support.md

```diff
@@ -0,0 +1,251 @@
+and the requirements defined in our requirements directory. Once these were installed I ran
+this tool https://pypi.org/project/checkmyreqs/ against all installed Python libraries. This brought to light
+the following libraries that will not support Python 3, which will impact some Salt components:
+
+ - pycassa (salt/modules/cassandra.py, salt/returners/cassandra_return.py)
+ - vbox (salt/modules/vbox_guest.py, salt/modules/vbox_guest.py,
+         salt/modules/vboxmanage.py, salt/utils/virtualbox.py,
+         salt/cloud/clouds/virtualbox.py)
+
+Both of these projects have not had commits added to the project in years. This will need to be documented
+for the Sodium release to make it clear that these modules will not work unless the upstream projects add
+Python 3 support.
+
+Going forward adding more test coverage will help to uncover these projects before we drop a Python version,
+but if we do not have enough test coverage, this exercise should be done any time a Python Version is dropped.
+There might be a conflict between a Salt dependency and its Python version support that will impact many users.
+If this conflict does occur we may need to delay dropping support for the End of Life Python version until that
+dependency is updated to the appropriate Python version.
+
+## Who is Affected
+
+Salt Contributors
+-----------------
+When a Python Version is dropped any contributions will need to ensure they align with the syntax
+of the lowest supported Python version. These contributors can look at Python's Release Notes for
+details around what is added into a release or removed. For example for Python 3.6
+https://docs.Python.org/3/whatsnew/3.6.html outlines some of the changes and removals added to this version.
+
+Salt User
+---------
+If a Salt user is using the packages from https://repo.saltstack.com they will not be affected
+as the Python binary will be included in the package. If they are pip installing the package and
+their operating system Python version is an End of Life Python version they will need to upgrade
+their Python version before being able to pip install Salt.
+
+Users creating their own packages
+---------------------------------
+If a user creates their own packages they will only be impacted if they are packaging for an operating
+system that includes an End of Life Python Version. They would need to either create their packages using
+salt-bin or include a supported version of Python in their packages.
+
+When will this happen?
+----------------------
+This Python Version Support change will go into affect as soon as it is approved, but will first impact
+the Sodium Release. The Sodium Release is due to be released in June and will support Python versions >= 3.5.
+
+## Alternatives
+[alternatives]: #alternatives
+
+- Alternative is to keep supporting insecure Python versions that line up with supported operating systems
```

Comment:
Supported operating systems will not have an insecure Python. The work that RedHat/Debian/etc. does is to ensure their LTS systems continue to have necessary security patches, even if the upstream developers don't provide them.

Reply:
not all of our supported operating systems follow this approach. I can change the syntax to say End of Life instead of insecure though to align with all of them.

Reply:
Which systems does Salt support, that do not support their own core packages?

Reply:
I could not see similar security support for macOS, aix and solaris. Please correct me if im wrong.

Reply:
I'd be very surprised if IBM and Oracle are leaving unpatched public vulnerabilities in their supported commercial systems. Looks like Apple have or are shortly going to stop packaging Python themselves.

Reply:
Yeah I honestly just couldn't find documentation. Either way I updated the phrase to be End of Life here instead of insecure.